File: scratch/test_reply.py
text = '[reply:{"id":"2213","sender":"Jinx","text":"تبا للنت"}]معلش يا حياتي ممواههه'
end_idx = text.find('}]')
print(f"end_idx: {end_idx}")
json_str = text[7:end_idx+1]
actual_text = text[end_idx+2:]
print(f"json_str: {json_str}")
print(f"actual_text: {actual_text}")

File: scratch/test_reply.js
function parseReplyMessage(raw){
  const text = String(raw || '').trim();
  if (text.startsWith('[reply:')) {
    const endIdx = text.indexOf('}]');
    if (endIdx !== -1) {
      const jsonStr = text.substring('[reply:'.length, endIdx + 1);
      const actualText = text.substring(endIdx + 2);
      try {
        const replyMeta = JSON.parse(jsonStr);
        return {
          isReply: true,
          replyId: replyMeta.id,
          replySender: replyMeta.sender,
          replyText: replyMeta.text,
          actualText: actualText
        };
      } catch (e) {
        console.warn('Failed to parse reply JSON:', e.message);
      }
    }
  }
  return { isReply: false, replyId: '', replySender: '', replyText: '', actualText: text };
}

const rawText = `[reply:{"id":"2213","sender":"Jinx","text":"تبا للنت"}]معلش يا حياتي ممواههه`;
const result = parseReplyMessage(rawText);
console.log('Result:', result);
